src/transformers/modeling_utils.py

```diff
@@ -1015,6 +1021,8 @@ def _generate_no_beam_search(
         for hypo_idx, hypo in enumerate(input_ids):
             decoded[hypo_idx, : sent_lengths[hypo_idx]] = hypo[: sent_lengths[hypo_idx]]
 
+        if self.config.is_encoder_decoder and decoder_start_token_id != bos_token_id:
+            return decoded[:, 1:]
```

Comment:
Same here, not a fan of having model specific behavior in the base class.
But if it's too much trouble to change it ok for now.